modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+   *   The current request object.
+   */
+  public function setPublicationDateOfComponent(string $publicationDateOfComponent) {
+    $this->publicationDateOfComponent = $publicationDateOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get article date.
+   *
+   * @return string|null
+   *   The article date.
+   */
+  public function getPublicationDateOfComponent() {
+    return $this->publicationDateOfComponent;
+  }
+
+  /**
+   * Set the article year.
+   *
+   * @param string $publicationYearOfComponent
+   *   Article year.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPublicationYearOfComponent(string $publicationYearOfComponent) {
+    $this->publicationYearOfComponent = $publicationYearOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the article year.
+   *
+   * @return string|null
+   *   The article year.
+   */
+  public function getPublicationYearOfComponent() {
+    return $this->publicationYearOfComponent;
+  }
+
+  /**
+   * Set article issue number.
+   *
+   * @param string $issueOfComponent
+   *   Article issue number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setIssueOfComponent(string $issueOfComponent) {
+    $this->issueOfComponent = $issueOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get article issue number.
+   *
+   * @return string
+   *    The issue number.
+   */
+  public function getIssueOfComponent() {
+    return $this->issueOfComponent;
+  }
+
+  /**
+   * Set the the volume number.
+   *
+   * @param string $volumeOfComponent
+   *   The article volume number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setVolumeOfComponent(string $volumeOfComponent) {
+    $this->volumeOfComponent = $volumeOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the article volume number.
+   *
+   * @return string|null
+   *   The article volume number.
+   */
+  public function getVolumeOfComponent() {
+    return $this->volumeOfComponent;
+  }
+
+  /**
+   * Set the page number in the publication.
+   *
+   * @param string $pagesOfComponent
+   *   The page number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPagesOfComponent(string $pagesOfComponent) {
+    $this->pagesOfComponent = $pagesOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the page number.
+   *
+   * @return string|null
+   *   The page number.
+   */
+  public function getPagesOfComponent() {
+    return $this->pagesOfComponent;
+  }
+
+  /**
+   * Send request to the service.
+   *
+   * @return bool
+   *   TRUE if request was send else FALSE.
+   */
+  public function send() {
+    $options = array(
+      'headers' => array(
+        'Content-Type' => 'application/xml; charset=UTF8',
+      ),
+      'body' => $this->buildMessage(),
+    );
+
+    try {
+      $client = new Client();
```

Comment:
In my opinion the client should be injected into the class - not created internally. This prevents unit testing of the class.

Reply:
Done